### PR TITLE
Miscellaneous fixes

### PR DESCRIPTION
**Pull request by salkinium**
Didn't want to open a PR for each of these small fixes.

cc @ekiwi.

### CODE REVIEW

File: src/xpcc/architecture/driver/accessor/flash_reader_avr_impl.hpp [hunk 1]
```diff
@@ -41,7 +41,7 @@
 #define	FLASH_STORAGE_STRING(s)			extern const char s[] PROGMEM; const char s[]
 #define	EXTERN_FLASH_STORAGE_STRING(s)	extern const char s[] PROGMEM
 
-#define	INLINE_FLASH_STORAGE_STRING(s)	((const PROGMEM char *)(s))
+#define	INLINE_FLASH_STORAGE_STRING(s)	PSTR(s)
```
Comment (ekiwi) [inline src/xpcc/architecture/driver/accessor/flash_reader_avr_impl.hpp:44]: is the `PROGMEM` attribute not needed here?

Comment (salkinium) [reply src/xpcc/architecture/driver/accessor/flash_reader_avr_impl.hpp:44]: No, for AVRs `PSTR` is provided by `avr-libc`.

Comment (ekiwi) [reply src/xpcc/architecture/driver/accessor/flash_reader_avr_impl.hpp:44]: ok, but why did you have to define it for other targets then?
I am assuming that `flash_reader_avr_impl.hpp` is only included for AVR targets, right?

Comment (salkinium) [reply src/xpcc/architecture/driver/accessor/flash_reader_avr_impl.hpp:44]: It's a replacement for `INLINE_FLASH_STORAGE_STRING`, so you can compile code for AVRs also on ARM Cortex-M.

Comment (ekiwi) [reply src/xpcc/architecture/driver/accessor/flash_reader_avr_impl.hpp:44]: but wasn't that the purpose of the `INLINE_FLASH_STORAGE_STRING` define, to provide a platform independent option?

Comment (salkinium) [reply src/xpcc/architecture/driver/accessor/flash_reader_avr_impl.hpp:44]: Yes, but:

``` cpp
XPCC_LOG_DEBUG << xpcc::accessor::asFlash(INLINE_FLASH_STORAGE_STRING("Temperature is ")) << t << xpcc::accessor::asFlash(INLINE_FLASH_STORAGE_STRING(" inside kitchen."));
```

OMGWTFBBQ was I smoking when I thought that `INLINE_FLASH_STORAGE_STRING` is an acceptable length for a macro that is to be used inline _a lot_?

Comment (ekiwi) [reply src/xpcc/architecture/driver/accessor/flash_reader_avr_impl.hpp:44]: hm, but `PSTR` isn't descriptive at all...

Comment (salkinium) [reply src/xpcc/architecture/driver/accessor/flash_reader_avr_impl.hpp:44]: No, but people coming from AVR will know it, and it won't break ARM Cortex-M this way.

**Pull request merged**